server/src/main/java/org/cloudfoundry/identity/uaa/util/UaaHttpRequestUtils.java

```diff
@@ -55,14 +68,33 @@ protected static ClientHttpRequestFactory createRequestFactory(HttpClientBuilder
     }
 
     protected static HttpClientBuilder getClientBuilder(boolean skipSslValidation) {
+        return getClientBuilder(skipSslValidation, 0,0, 0);
+    }
+
+    protected static HttpClientBuilder getClientBuilder(boolean skipSslValidation, int poolSize, int defaultMaxPerRoute, int maxKeepAlive) {
         HttpClientBuilder builder = HttpClients.custom()
             .useSystemProperties()
             .setRedirectStrategy(new DefaultRedirectStrategy());
         if (skipSslValidation) {
             builder.setSslcontext(getNonValidatingSslContext());
             builder.setSSLHostnameVerifier(new NoopHostnameVerifier());
         }
-        builder.setConnectionReuseStrategy(NoConnectionReuseStrategy.INSTANCE);
+        if (poolSize > 0) {
```

Comment:
I think it would be simpler to provide reasonable defaults and then always build and configure the `PoolingHttpClientConnectionManager`, reducing all of these branches.

Reply:
I implemented it this way to guarantee, that not configuring the values would not change anything. But if you prefer it with such a simpler structure, I will update the code like this.